cloudinit/node.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinit
+
+const (
+	nodeCloudInit = `{{.Header}}
+write_files:
+-   path: /tmp/kubeadm-node.yaml
+    owner: root:root
+    permissions: '0640'
+    content: |
+      ---
+{{.JoinConfiguration | Indent 6}}
+kubeadm:
+  operation: join
+  config: /tmp/kubeadm-node.yaml
+`
+)
+
+// NodeInput defines the context to generate a node user data.
+type NodeInput struct {
+	baseUserData
+
+	JoinConfiguration string
+	AdditionalFiles   []Files
```

Comment:
AdditionalFiles/Commands could probably be moved into baseUserData?

Reply:
That makes sense to me

Reply:
You'll need to export/expose `baseUserData` to make the DX a little better otherwise this fails:

```
input := userdata.NodeInput{
    AdditionalFiles: []userdata.Files{[...]}, 
    [...]
}
```

Users of this package will need to do something like
```
input := userdata.NodeInput{
    [...]
}
input.AdditionalFiles = []userdata.Files{[...]}
`

Reply:
Ah you're totally right. Will fix.